website/pages/blog/graphql-eslint-3.14.mdx

```diff
@@ -0,0 +1,329 @@
+dependency to be installed.
+
+<Callout>
+  Also, GraphQL Config is no longer bundled with `cosmiconfig-toml-loader` and
+  `cosmiconfig-typescript-loader`. You must install it manually in case of using TOML or TypeScript
+  config. The benefit of this is that bundle size is reduced by 35%.
+</Callout>
+
+## New Rules
+
+The new version introduces 4 new rules:
+
+### Rule `require-nullable-fields-with-oneof`
+
+Require `input` or `type` fields to be non-nullable with `@oneOf` directive
+
+```json filename=".eslintrc.json"
+{
+  "rules": {
+    "@graphql-eslint/require-nullable-fields-with-oneof": "error"
+  }
+}
+```
+
+<Comparison>
+```graphql filename="❌ Incorrect" copy=false
+input Input @oneOf {
+  foo: String!
+  bar: [Int!]!
+}
+```
+```graphql filename="✅ Correct" copy=false
+input Input @oneOf {
+  foo: String
+  bar: [Int!]
+}
+```
+</Comparison>
+
+### Rule `require-type-pattern-with-oneof`
+
+Enforce types with `@oneOf` directive have `error` and `ok` fields.
+
+- It's easier to communicate user errors in the response
+- Errors can contain any additional info the client might need (error code, validation info and so
+  on)
+- Reduce the need to use `... on Error` and so on, no need to specify type names in the query
+
+```json filename=".eslintrc.json"
+{
+  "rules": {
+    "@graphql-eslint/require-type-pattern-with-oneof": "error"
+  }
+}
+```
+
+<Comparison>
+```graphql filename="❌ Incorrect" copy=false
+type DoSomethingMutationResult @oneOf {
+  notok: DoSomethingSuccess
+  noterror: Error
+}
+```
+```graphql filename="✅ Correct" copy=false
+type DoSomethingMutationResult @oneOf {
+  ok: DoSomethingSuccess
+  error: Error
+}
+```
+</Comparison>
+
+### Rule `lone-executable-definition`
+
+Require queries, mutations, subscriptions or fragments to be located in separate files.
+
+```json filename=".eslintrc.json"
+{
+  "rules": {
+    "@graphql-eslint/lone-executable-definition": "error"
+  }
+}
+```
+
+<Comparison>
+```graphql filename="❌ Incorrect" copy=false
+# users.graphql
+query Users {
+  users {
+    ...UserFields
+  }
+}
+fragment UserFields on User {
+  id
+}
+```
+```graphql filename="✅ Correct" copy=false
+# users.graphql
+query Users {
+  users {
+    ...UserFields
+  }
+}
+```
+</Comparison>
+
+### Rule `no-one-place-fragments`
+
+An original proposal for a rule to suggest inline fragments that are spread only in one place was
+asked [2 years ago](https://github.com/B2o5T/graphql-eslint/issues/72#issue-715054179) and finally,
+this rule is a part of GraphQL-ESLint.
+
+```json filename=".eslintrc.json"
+{
+  "rules": {
+    "@graphql-eslint/no-one-place-fragments": "error"
+  }
+}
+```
+
+<Comparison>
+```graphql filename="❌ Incorrect" copy=false
+fragment UserFields on User {
+  id
+}
+
+{
+  user {
+    ...UserFields
+  }
+}
+```
+```graphql filename="✅ Correct" copy=false
+{
+  user {
+    ...UserFields
+    friends {
+      ...UserFields
+    }
+  }
+}
+```
```

Comment:
I'd add another example to better indicate the behavior:
```graphql filename="✅ Correct" copy=false
{
  user {
    id
  }
}
```
(adding commit suggestion fails because of snippet inside snippet 🤦‍♂️)

Reply:
I think is fine because I wanted to show count of fragment spreads in examples